File: project_euler/reverse_bits.cpp
#include <bits/stdc++.h>
using namespace std;
long long int solution(long long int n){
    long long count = 0;
    long long int ans =0;
    while(n>0){
        count++;
        n = n>>1;
        cout<<n<<endl;
    }
    return ans;
}
int main(){
    long long int n;
    cin>>n;
    long long int ans = solution(n);
    //cout<<ans<<endl;
}

File: project_euler/reverse_bits.py
def solution(n):
    count =0
    ans = 0
    while(n>0):
        count+=1
        if(n&1):
            temp = 1<<(32-count)
            ans+=temp 
        n = n>>1
    return ans

n  = int(input())
ans = solution(n)
print(ans,end="\n")
